src/code42cli/cmds/search/extraction.py

```diff
@@ -83,11 +75,10 @@ def handle_response(response):
         total_events = len(events)
         handlers.TOTAL_EVENTS += total_events
 
-        for events_per_page in _paginate(events, _EVENT_COUNT_PER_PAGE):
-            output = _process_events(
-                output_format, include_all, events_per_page, output_header
-            )
-            click.echo_via_pager(output)
+        def paginate():
```

Comment:
I like this!

Note to others:  Kiran and I were on a call and found out that by passing the generator in, `echo_via_page()` will handle pagination for us. Meaning, it display records to fit in the current terminal size and then hitting enter allows you to page through them.

Reply:
Welll.... this isn't quite enough.... So, the END paging happens once the generator runs outs..... UGH!


This is a for sure finding:   The way we have the extraction handlers set up does not plug in well to `echo_via_pager()`

Reply:
OK maybe it is not exiting early, it just does the END thing multiple times, which is not ideal, but in that case it is fine to have the headers multiple times.

Reply:
I don't see paging past 10K events in the terminal being a common occurrence so I wouldn't worry about this at the moment.

Reply:
Yeah, since `handle_response` is expected to do all the actual processing of the events calling `echo_via_pager()` multiple times will give us a pager process for each page, instead of it appending it all into a single pager process. 

I think our only option would be to have `extract()` be called on a separate thread, and `handle_response()` append the results to a queue. Then we could on the main thread have a generator function that pulled events off the queue as they came in to pass to `echo_via_pager`. I think that would get us updates in the pager in "real time" while being able to pass everything in a single generator.